samwise.js

```diff
@@ -1,140 +1,142 @@
 /**
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.